lib/eunit/src/eunit_proc.erl

```diff
@@ -273,7 +273,9 @@ kill_task(Child, St) ->
     terminate_insulator(St).
 
 buffer_to_binary([B]) when is_binary(B) -> B;  % avoid unnecessary copying
-buffer_to_binary(Buf) -> list_to_binary(lists:reverse(Buf)).
+buffer_to_binary(Buf) ->
```

Comment:
I'm not sure I can explain it, but it works in all cases (i.e. even with mixed encodings) for the use cases I could come up with. 

Logically, the only way it could work is if the input is either a unicode list or a list of already utf8-encoded unicode. Buffer_to_binary is only used with input that comes from the test's output, so my guess is that the io loop takes care of converting the data to that format.

I tried a lot of different alternatives, but didn't manage to get any other that would cover all cases. Either latin1 slips through, or utf8 gets double encoded. I tried to reorganize the code from the ground, so that it's clear which encoding is applied where, but it got too intrusive. If this half-good solution is not acceptable, then I can go for the better one, but I can't say when it will be ready.